projects/train/train/data/waveforms/generator/cbc.py

```diff
@@ -9,12 +9,37 @@
 
 class FrequencyDomainCBCGenerator(WaveformGenerator):
     """
-    A torch module for generating CBC waveforms on the fly using
-    ml4gw waveform models.
+    A torch module for generating waveforms on the fly.
 
     Args:
+        waveform:
+            A callable that returns cross and plus polarizations
+            given a set of parameters.
+        duration:
+            The duration of the waveform in seconds
+        sample_rate:
+            Sample rate of the waveform in Hz
+        f_min:
+            The minimum frequency of the waveform in Hz
+        f_max:
+            The maximum frequency of the waveform in Hz
+        padding:
+            The amount of zero padding
+            to add to the right of the waveform in seconds.
+            The coalescence time of the waveform is automatically
+            placed 0.5 seconds from the right edge to account for
+            the ringdown. This parameter will add additional
+            padding on top of that.
+        waveform_arguments:
+            A dictionary of fixed arguments to pass to the waveform model,
+            e.g. `f_ref` for CBC waveforms
     """
 
+    # duration of ringdown - will ensure
+    # coalescence time is at least this
+    # far from the right edge
+    RINGDOWN_DURATION = 0.5
```

Comment:
Let's make this configurable with a default of 0.5s

Reply:
The main reason I decided not to make it configurable in the `__init__` was to protect users from fiddling with it and breaking things.

Reply:
yeah but the ringdown is different based on the priors, similar to f_max. You can add a note in the docs about it if you like to.

Reply:
done